pyathena/sqlalchemy/base.py

```diff
@@ -1103,6 +1103,15 @@ def has_table(
         except NoSuchTableError:
             return False
 
+    @reflection.cache
+    def get_view_definition(self, connection, view_name: str, schema: str, **kw):
+        query = f"""SHOW CREATE VIEW "{schema}"."{view_name}";"""
+        res = connection.scalars(text(query))
+        if res is None:
```

Comment:
If the View does not exist, the query result will not be None, but an OperationalError will be generated. I do not think that all cases where an OperationalError occurs are cases where the View does not exist, but if an OperationalError occurs, I think it is fine to generate a NoSuchTableError.

Reply:
I have followed the implementation of `sqlalchemy` here to stay consistent. See these two examples for this:
* [Postgres](https://github.com/sqlalchemy/sqlalchemy/blob/7cf836241aa72494b42b645a3ddec04b0a6e6061/lib/sqlalchemy/dialects/postgresql/base.py#L3438)
* [Oracle](https://github.com/sqlalchemy/sqlalchemy/blob/7cf836241aa72494b42b645a3ddec04b0a6e6061/lib/sqlalchemy/dialects/oracle/base.py#L3136)

Reply:
Does the same OperationalError occur in Oracle or Postgres when the View does not exist? 🤔

Reply:
Ah, now I see what you mean 😅. 
You are right, I should probably get rid of `if res is None` and replace it with a `try: ... except OperationalError` statement.